creator-node/src/routes/playlists.js

```diff
@@ -0,0 +1,182 @@
+        )
+      }
+
+      // Record metadata file entry in DB
+      const transaction = await models.sequelize.transaction()
+      let fileUUID
+      try {
+        const createFileQueryObj = {
+          multihash,
+          sourceFile: req.fileName,
+          storagePath: dstPath,
+          type: 'metadata'
+        }
+        const file = await DBManager.createNewDataRecord(
+          createFileQueryObj,
+          cnodeUserUUID,
+          models.File,
+          transaction
+        )
+        fileUUID = file.fileUUID
+        await transaction.commit()
+      } catch (e) {
+        await transaction.rollback()
+        return errorResponseServerError(
+          `Could not save playlist metadata to db: ${e}`
+        )
+      }
+
+      // This call is not await-ed to avoid delaying or erroring
+      issueAndWaitForSecondarySyncRequests(req)
+
+      return successResponse({
+        metadataMultihash: multihash,
+        metadataFileUUID: fileUUID
+      })
+    })
+  )
+
+  /**
+   * Given playlist playlistId, blockNumber, and metadataFileUUID, creates/updates Playlist DB entry
+   * and associates image file entries with playlist. Ends playlist creation/update process.
+   */
+  app.post(
+    '/playlists',
+    authMiddleware,
+    ensurePrimaryMiddleware,
+    ensureStorageMiddleware,
+    syncLockMiddleware,
+    handleResponse(async (req, res) => {
+      const { playlistId, blockNumber, metadataFileUUID } = req.body
+
+      if (!playlistId || !blockNumber || !metadataFileUUID) {
+        return errorResponseBadRequest(
+          'Must include playlistId, blockNumber, and metadataFileUUID.'
+        )
+      }
+
+      const cnodeUser = req.session.cnodeUser
+      if (blockNumber < cnodeUser.latestBlockNumber) {
+        return errorResponseBadRequest(
+          `Invalid blockNumber param ${blockNumber}. Must be greater or equal to previously processed blocknumber ${cnodeUser.latestBlockNumber}.`
+        )
+      }
+
+      const cnodeUserUUID = req.session.cnodeUserUUID
+
+      // Fetch metadataJSON for metadataFileUUID.
+      const file = await models.File.findOne({
+        where: { fileUUID: metadataFileUUID, cnodeUserUUID }
+      })
+      if (!file) {
+        return errorResponseBadRequest(
+          `No file db record found for provided metadataFileUUID ${metadataFileUUID}.`
+        )
+      }
+      let metadataJSON
+      try {
+        const fileBuffer = await readFile(file.storagePath)
+        metadataJSON = JSON.parse(fileBuffer)
+      } catch (e) {
+        return errorResponseServerError(
+          `No file stored on disk for metadataFileUUID ${metadataFileUUID} at storagePath ${file.storagePath}: ${e}.`
+        )
+      }
+
+      // Get coverArtFileUUID for multihashes in metadata object, if present.
+      let coverArtFileUUID
+      try {
+        coverArtFileUUID = await validateStateForImageDirCIDAndReturnFileUUID(
+          req,
+          metadataJSON.cover_photo_sizes
```

Comment:
i think this needs to be `playlist_image_sizes_multihash` based on https://github.com/AudiusProject/audius-protocol/blob/PLAT-112_upload_playlist_metadata/discovery-provider/src/models/models.py#L383